From 685beb8e962d7e7a0b1a7b5aed1a4669308141cd Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Tue, 24 Mar 2026 18:51:21 +0000
Subject: [PATCH 03/12] net: dsa: mxl862xx: warn about old firmware default PCE
 rules

Firmware versions older than 1.0.80 install global PCE rules at
boot that redirect link-local frames (BPDUs, LLDP, LACP) to port 0
(the on-chip microcontroller) instead of the DSA CPU port.  With
port 0 disabled under DSA, these rules silently drop matching
traffic.

Log a warning when old firmware is detected so users know to update.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 drivers/net/dsa/mxl862xx/mxl862xx.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/dsa/mxl862xx/mxl862xx.c
+++ b/drivers/net/dsa/mxl862xx/mxl862xx.c
@@ -998,6 +998,10 @@ static int mxl862xx_setup(struct dsa_swi
 	if (ret)
 		goto free_trap_fid;
 
+	if (!MXL862XX_FW_VER_MIN(priv, 1, 0, 80))
+		dev_warn(ds->dev, "firmware < 1.0.80 installs global PCE rules "
+			 "that interfere with DSA operation, please update\n");
+
 	ret = mxl862xx_setup_mdio(ds);
 	if (ret)
 		goto free_trap_fid;
